tests/Unit/Framework/Pypp.hpp

```diff
@@ -209,6 +211,50 @@ struct ContainerPackAndUnpack<std::tuple<Tensor<T, Symm, Indices>...>,
   }
 };
 
+template <typename... Tags, typename ConversionClassList>
+struct ContainerPackAndUnpack<tuples::TaggedTuple<Tags...>, ConversionClassList,
+                              std::nullptr_t> {
+ private:
+  using first_tag = tmpl::front<tmpl::list<Tags...>>;
+
+ public:
+  template <typename Tag>
+  struct UnpackedTag {
+    static std::string name() { return db::tag_name<Tag>(); }
+    using type = typename ContainerPackAndUnpack<
+        typename Tag::type, ConversionClassList>::unpacked_container;
+  };
+  using unpacked_container = tuples::TaggedTuple<UnpackedTag<Tags>...>;
+  using packed_container = tuples::TaggedTuple<Tags...>;
+  // This assumes the first member of the TaggedTuple sets the packed
+  // type. I'm (Nils Deppe) not sure how else to handle this right now.
+  using packed_type =
```

Comment:
so does this mean the tagged tuple must have the same packed_type for each element?  Is there a way to static_assert this?

Reply:
I don't know how to check it (easily?) since other types are okay, as long as they aren't packed. I.e. if you want to pass in or out an int or double that's fine, it just can't be the first entry in the tuple. I don't know exactly how to assert this check since it's not different packed types that are bad.

Reply:
okay, then I am fine with the PR being merged